src/databricks/labs/ucx/hive_metastore/mapping.py

```diff
@@ -103,10 +116,21 @@ def current_tables(self, tables: TablesCrawler, workspace_name: str, catalog_nam
         for table in tables_snapshot:
             yield Rule.initial(workspace_name, catalog_name, table, self._recon_tolerance_percent)
 
+    @staticmethod
+    def tables_not_mapped(tables_crawler: TablesCrawler, current_tables: list[Rule], workspace_name: str):
+        crawled_tables_keys = [crawled_table.key for crawled_table in tables_crawler.snapshot()]
+        hms_table_keys = [rule.as_hms_table_key for rule in current_tables]
+        for crawled_table_key in crawled_tables_keys:
+            if crawled_table_key not in hms_table_keys:
+                yield TableNotMapped.initial(workspace_name, crawled_table_key)
```

Comment:
```suggestion
                yield TableNotMapped(workspace_name, crawled_table_key)
```
remove useless `initial` method.

Reply:
The implementation approach has been changed.